Buildings/Fluid/HeatExchangers/BaseClasses/VariableSpeedThermalWheels/Sensible.mo

```diff
@@ -0,0 +1,99 @@
+    annotation (Dialog(group="Efficiency"));
+  parameter whe.BaseClasses.Data.Generic per
+    "Record with performance data"
+    annotation (Placement(transformation(extent={{60,60},{80,80}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uSpe(
+    final unit="1",
+    final max=1)
+    "Wheel speed ratio"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+        iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput P(final unit="W")
+    "Electric power consumption"
+    annotation (Placement(transformation(extent={{100,60},{140,100}}),
+        iconTransformation(extent={{100,60},{140,100}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput epsSenCor(final unit="1")
+    "Sensible heat exchanger effectiveness correction"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}}),
+    iconTransformation(extent={{100,-20},{140,20}})));
+
+initial equation
+  for i in 1:size(yeta,1)-1 loop
+         assert(xSpe[i]/yeta[i] < 1 + 1E-4,
+                  "In " + getInstanceName() + ": the motor efficiency curve is wrong. 
+                  The ratio of the speed ratio to the motor percent 
+                  full-load efficiency should be less than 1",
+                  level=AssertionLevel.error)
+                  "Check if the motor efficiency curve is correct";
+  end for;
+  assert(abs(yeta[size(yeta,1)]-1) < 1E-4,
+          "In " + getInstanceName() + ": motor efficiency curve is wrong. 
+          The motor percent full-load efficiency at the full seepd should be 1",
+          level=AssertionLevel.error)
+          "Check if the motor efficiency curve is consistent with the nominal condition";
+equation
+  P = per.P_nominal*uSpe/Buildings.Utilities.Math.Functions.smoothInterpolation(
+        x=uSpe,
+        xSup=xSpe,
+        ySup=yeta)
+        "Calculate the wheel power consumption";
+  epsSenCor = Buildings.Utilities.Math.Functions.smoothInterpolation(
+                x=uSpe,
+                xSup=per.senHeatExchangeEffectiveness.uSpe,
+                ySup=per.senHeatExchangeEffectiveness.epsCor)
+                "Calculate the sensible heat exchanger effectiveness correction";
+   annotation (
+   defaultComponentName="senWhe",
+   Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
+        coordinateSystem(preserveAspectRatio=false)),
+    Documentation(info="<html>
+<p>
+This model calculates the power consumption and the sensible heat exchanger 
+effectiveness correction of a sensible heat wheel.
+Specifically, this calculation is configured as follows.
+</p>
+<ul>
+<li>
+The power consumption of this wheel is calculated by
+<p align=\"center\" style=\"font-style:italic;\">
+P = P_nominal * uSpe / eta,
+</p>
+<p>
+where <code>P_nominal</code> is the nominal wheel power consumption,
+<code>uSpe</code> is the wheel speed ratio, 
+and the <code>eta</code> is the motor percent full-load efficiency, i.e.,
```

Comment:
need a clear explanation of the `eta`, what is it? is it `eff(uSpe=x) / eff(uSpe=1), where the `eff(uSpe=x)` is the motor efficiency when the speed ratio is `x`?

Reply:
Added the detailed explanation as suggested.